.github/workflows/ci.yml

```
@@ -21,6 +19,37 @@ jobs:
       - name: restore build & cargo cache
         uses: Swatinem/rust-cache@v1
 
+      - name: Build
+        run: cargo build --workspace --locked
+
+      - name: compile test binaries
+        # source:
+        # https://github.com/rust-lang/cargo/issues/1924#issuecomment-289770287
+        run: "
+          mkdir ./test-binaries/ &&
+          cargo test --no-run --message-format=json |
+          jq -r \"select(.profile.test == true) | .filenames[]\" |
+          head -n 1 |
+          xargs -I _ cp _ ./test-binaries/docs-rs-tests
+          "
```

I have a suspicion this will run into https://gitlab.com/gitlab-org/gitlab-runner/-/issues/25460 - let's use separate commands for each instead.
```suggestion
        run: |
          mkdir ./test-binaries
          cargo test --no-run --message-format=json |
            jq -r "select(.profile.test == true) | .filenames[]" |
            head -n 1 |
            xargs cp --target ./test-binaries/docs-rs-tests
```
Also, don't you need to add backslashes after each line?